newIDE/app/src/MainFrame/index.js

```diff
@@ -537,6 +537,8 @@ const MainFrame = (props: Props) => {
             ),
             error
           );
+          setIsLoadingProject(false);
+          return Promise.reject(error);
```

Comment:
`setIsLoadingProject(false)` brings the app to a usable state, otherwise, it remains in the loading state unless another project is opened.
Also, the error is propagated to remove the sad path file from localStorage.

Reply:
>  usable state, otherwise, it remains in the loading state unless another project is opened.

Then this is still an inconsistent state right? We should remove the loading state once the loading is done, be it successfully or with a failure.

Reply:
> Then this is still an inconsistent state right?

I'm not sure if it's really an inconsistent state, but we should stop loading after a failure, right?

Reply:
> EDIT: Not being able to open a sad path should be a failure, no?

Yes. 
Basically opening a project is either successful (project is opened, loading is stop) or not (for whatever reason: file not present, you don't have the permission, a meteor destroyed your hard drive...), in which case project is not opened, a message is shown and then the loading should stop - because you're back at the state where you can try to re-open your project or do something else.

Reply:
Alright @4ian, so the loading is not stopped until the project is opened successfully. Basically, when this error comes up, the app remains in a loading state after this failure.

> it remains in the loading state unless another project is opened.

When we successfully open another project, loading stops. So I believe, the loading state wasn't handled properly for failures (maybe during the refactor?).

Reply:
> So I believe, the loading state wasn't handled properly for failures (maybe during the refactor?).

Possible. Can you have a look at this? If it's too complicated let me know and we'll fix this in master first?

Reply:
@4ian, for successful openings, `openLayout` sets the loading state to false, (or `openSceneOrProjectManager` sets it false if the project has multiple scenes).

Now, there are two failures, one of them is the sad path, which is handled correctly now (and also a bit extended to remove these sad paths from localStorage). The other one occurs after `checkForAutosaveAfterFailure()`, I don't really know how to reproduce this.

Reply:
I fixed the failure-not-reset-loading-state issue in master. Can you give a try to merging/rebasing your PR on master? :) Thanks!